### ISSUE TITLE
missing intersect_bvh() implementation

### ISSUE THREAD

**Issue by voxel-tracer** [xelatihy/yocto-gl#1263]
`yocto_bvh.hpp` defines the following method:
```ccp
bvh_intersection intersect_bvh(const bvh_shape& bvh, const scene_shape& shape, const ray3f& ray, bool find_any = false, bool non_rigid_frames = true);
```
but the only implementation I could find is the following in `yocto_bvh.cpp`:
```ccp
static bool intersect_bvh(const bvh_shape& bvh, const scene_shape& shape, const ray3f& ray_, int& element, vec2f& uv, float& distance, bool find_any)
```

**Comment by xelatihy** [comment]
You are right. I fixed this in a branch for now. You can use the code below to get the intersection.

```cpp
bvh_intersection intersect_bvh(const bvh_data& bvh, const shape_data& shape,
    const ray3f& ray, bool find_any) {
  auto intersection = bvh_intersection{};
  intersection.hit  = intersect_bvh(bvh, shape, ray, intersection.element,
      intersection.uv, intersection.distance, find_any);
  return intersection;
}
```

Note though that in the next release I will change th signature of this method to return a `shape_intersection` that does not contain an `instance` variable.

**Comment by xelatihy** [comment]
Fixed in #1264
